### PR TITLE
add options to specify which search filters to show

### PR DESCRIPTION
**Pull request by chadwickmeyer**
options: "location", "basic", "advanced"
add option to show a link to the advanced search (used on modules that embed this component)
Add option to show advanced search link on list component as well (needed if there is no search bar)

### CODE REVIEW

**Comment by chadwickmeyer** [comment]
NOTE: @alexgurrola @apocist  I tried to run `yarn test` and it failed. So this may not be properly formatted or tested but I think these are problems with other components and files.

 Finished 'compileLESS' after 661 ms
packages/angular/src/tree/tree.component.ts(41,24): error TS2307: Cannot find module '@agentepsilon/decko' or its corresponding type declarations.
packages/angularjs-extras/src/filters/numeral.ts(12,21): error TS2307: Cannot find module 'numeral' or its corresponding type declarations.
TypeScript: 2 semantic errors
TypeScript: emit succeeded (with errors)
[15:53:04] 'compileTypeScript' errored after 15 s
[15:53:04] Error: TypeScript: Compilation failed
    at Output.mightFinish (/Users/chadwickmeyer/Sites/stratus/node_modules/gulp-typescript/release/output.js:130:43)
    at /Users/chadwickmeyer/Sites/stratus/node_modules/gulp-typescript/release/output.js:43:22
[15:53:04] 'compile' errored after 15 s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by apocist** [comment]
> NOTE: @alexgurrola @apocist I tried to run `yarn test` and it failed. So this may not be properly formatted or tested but I think these are problems with other components and files.
> 
> Finished 'compileLESS' after 661 ms
> packages/angular/src/tree/tree.component.ts(41,24): error TS2307: Cannot find module '@agentepsilon/decko' or its corresponding type declarations.
> packages/angularjs-extras/src/filters/numeral.ts(12,21): error TS2307: Cannot find module 'numeral' or its corresponding type declarations.
> TypeScript: 2 semantic errors
> TypeScript: emit succeeded (with errors)
> [15:53:04] 'compileTypeScript' errored after 15 s
> [15:53:04] Error: TypeScript: Compilation failed
> at Output.mightFinish (/Users/chadwickmeyer/Sites/stratus/node_modules/gulp-typescript/release/output.js:130:43)
> at /Users/chadwickmeyer/Sites/stratus/node_modules/gulp-typescript/release/output.js:43:22
> [15:53:04] 'compile' errored after 15 s
> error Command failed with exit code 1.
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

That sounds like you need to locally run `yarn install` to download all the modules that you need to compile (to allow testing)

File: packages/idx/src/property/search.component.ts [hunk 1]
```diff
@@ -44,6 +44,10 @@ export type IdxPropertySearchScope = IdxSearchScope & {
     listId: string
     listInitialized: boolean
     listLinkUrl: string
+    // Can be 'simple' (location only), 'basic' (beds/baths, price, etc), 'advanced' (dropdown filters)
+    searchType: string
+    advancedSearchUrl: boolean
```
Comment (apocist) [inline packages/idx/src/property/search.component.ts:49]: due to `data-ng-href="advancedSearchUrl"` this should be a `string`

File: packages/idx/src/property/search.component.ts [hunk 2]
```diff
@@ -112,7 +119,9 @@ Stratus.Components.IdxPropertySearch = {
             $scope.listInitialized = false
             $scope.listLinkUrl = $attrs.listLinkUrl || '/property/list'
             $scope.listLinkTarget = $attrs.listLinkTarget || '_self'
-
+            $scope.searchType = $attrs.searchType || 'advanced'
+            $scope.advancedSearchUrl = $attrs.advancedSearchUrl || true
```
Comment (apocist) [inline packages/idx/src/property/search.component.ts:123]: due to `data-ng-href="advancedSearchUrl"` this should default to `''`

**Review by apocist — APPROVED** [review]
I'll this merged and and running with the new options

**Pull request merged**